lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php

```diff
@@ -123,41 +126,78 @@ public function convertException($message, DriverException $exception)
 
     /**
      * {@inheritdoc}
+     *
+     * @return AbstractPlatform|MariaDb102Platform|MySQL57Platform|MySqlPlatform
+     * @throws DBALException
      */
-    public function createDatabasePlatformForVersion($version)
+    public function createDatabasePlatformForVersion($version) : AbstractPlatform
     {
-        if ( ! preg_match('/^(?P<major>\d+)(?:\.(?P<minor>\d+)(?:\.(?P<patch>\d+))?)?/', $version, $versionParts)) {
+        if (false !== stripos($version, 'mariadb')) {
```

Comment:
This whole block could be simplified a bit:

```php
if (false !== stripos($version, 'mariadb')
    && version_compare($this->getMariaDbMysqlVersionNumber($version), '10.2.7', '>=')) {
    return new MariaDb102Platform();
}

if (version_compare($this->getOracleMysqlVersionNumber($version), '5.7.9', '>=')) {
    return new MySQL57Platform();
}

return $this->getDatabasePlatform();
```

Reply:
Mmmm, not doing the same thing. What if the version is `10.0.15-MariaDB-1~wheezy`. In this case the `MySQL57Platform()` will be incorrectly chosen instead of `MysqlPlatform()`. An equivalent code would be:

```php
        if (false !== stripos($version, 'mariadb')
            && version_compare($this->getMariaDbMysqlVersionNumber($version), '10.2.7', '>=')) {
            return new MariaDb102Platform();
        }

        if (false === stripos($version, 'mariadb')
            && version_compare($this->getOracleMysqlVersionNumber($version), '5.7.9', '>=')) {
            return new MySQL57Platform();
        }

        return $this->getDatabasePlatform();
```